app/services/burn-event/service.go

```diff
@@ -108,13 +115,124 @@ func (s Service) ProcessEvent(event transfer.Transfer) {
 		return
 	}
 
-	for _, splitTransfer := range splitTransfers {
+	countOfTransfers := len(splitTransfers)
+	// Wait Group and resultPerTransfer slice are needed to wait for the end result of the mined transactions
+	// to set metric value.
+	wg := new(sync.WaitGroup)
+	wg.Add(countOfTransfers)
+	resultPerTransfer := make([]*bool, countOfTransfers)
+	receiverFound := false
```

Comment:
Given that the splitTransfers always contains the receiver transfer, this will always be true?

Reply:
We have true only if all results are true.

Reply:
One of the splitTransfers will always that contains the receiver, there is no point in having this boolean.